distributed/diagnostics/nvml.py

```diff
@@ -1,19 +1,31 @@
 import pynvml
 
-pynvml.nvmlInit()
-count = pynvml.nvmlDeviceGetCount()
+need_pynvml_init = True
+handles = None
 
-handles = [pynvml.nvmlDeviceGetHandleByIndex(i) for i in range(count)]
+
+def _initialize_pynvml():
+    global need_pynvml_init, handles
+    pynvml.nvmlInit()
+    count = pynvml.nvmlDeviceGetCount()
+    handles = [pynvml.nvmlDeviceGetHandleByIndex(i) for i in range(count)]
+    need_pynvml_init = False
 
 
 def real_time():
+    global need_pynvml_init, handles
+    if need_pynvml_init:
+        _initialize_pynvml()
```

Comment:
I wonder if we might include the check inside the `_initialize_pynvml` call.  This function would then just become idempotent.  We would call it all the time, and it would decide whether or not it needed to call the `pynvml` function.

Reply:
yes, and if that's done, I'd prefer to change the name to `_ensure_pynvml_initialized()` to indicate that it may not (re)run initialization.